eth/blockwatch/block_watcher.go

```diff
@@ -206,14 +206,36 @@ func (w *Watcher) pollNextBlock() error {
 	// Even if an error occurred, we still want to emit the events gathered since we might have
 	// popped blocks off the Stack and they won't be re-added
 	if len(events) != 0 {
-		w.blockFeed.Send(events)
+		w.blockFeed.Send(w.enrichWithL1(events))
 	}
 	if err != nil {
 		return err
 	}
 	return nil
 }
 
+// Add L1 Block Number to the event with the highest block number
+func (w *Watcher) enrichWithL1(events []*Event) []*Event {
```

Comment:
Could be good to have some test cases for this.

Reply:
Added.